webcompat/views.py

```diff
@@ -153,6 +154,10 @@ def file_issue():
 @app.route('/', methods=['GET'])
 def index():
     """Set the main view where people come to report issues."""
+    push('/css/dist/webcompat.min.css', **{
+        'as': 'style',
+        'rel': 'preload'
+    })
```

Comment:
@marimeireles 
So if I look at the results locally. This does the right thing in sending the right headers.

```http
HTTP/1.0 200 OK
Content-Type: text/html; charset=utf-8
Content-Length: 51341
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
X-Frame-Options: DENY
Content-Security-Policy: default-src 'self'; object-src 'none'; connect-src 'self' https://api.github.com; font-src 'self' https://fonts.gstatic.com; img-src 'self' https://webcompat.com https://www.google-analytics.com https://*.githubusercontent.com data:; manifest-src 'self'; script-src 'self' https://www.google-analytics.com https://api.github.com 'nonce-63c75e1bc1745d52ccbe69eca1c683ba2795a918'; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com; base-uri 'self'; frame-ancestors 'self'; report-uri /csp-report
Link: </css/dist/webcompat.min.css>; as=style; rel=preload,
Vary: Cookie
Set-Cookie: session=******; HttpOnly; Path=/
Server: Werkzeug/0.14.1 Python/2.7.10
Date: Wed, 06 Feb 2019 02:08:25 GMT
```

We can see `Link: </css/dist/webcompat.min.css>; as=style; rel=preload,` at the HTTP level.

We probably need to check if the trailing comma could create an issue too. 

On the other hand it does nothing to the HTML side of things, where the link stays the same.

```html
<link href="/css/dist/webcompat.min.css" type="text/css" rel="stylesheet">
```

So I wonder if the issue is not just that. Aka the HTML Link overriding the HTTP Link header.

MDN docs doesn't mention HTTP in [Link preload](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content), just the HTML part.

I don't see anything in the [processing model of preload spec](https://w3c.github.io/preload/#processing) except the first step.

> When the user agent that supports [RFC5988] creates a Document and processes `Link` headers that contain a preload link. 

Though it says

> The user agent SHOULD abort the current request if the `href` attribute of the link element of a preload link is changed, removed, or its value is set to an empty string.

but this is only for changing the `href`, not other values.

So I guess if the HTML parsing says something different, I wonder if it cancels the instructions from the HTTP `Link`

Checking [RFC8288](https://tools.ietf.org/html/rfc8288#appendix-A.1) which obsoleted RFC5988.

And [HTML is not clear either ](https://html.spec.whatwg.org/multipage/semantics.html#processing-link-headers) about possible conflicts.

yoohoo there is another spec talking about it [CSSOM](https://drafts.csswg.org/cssom/#requirements-on-user-agents-implementing-the-http-link-header).

Reply:
>1. remove from the HTML template to see if it's working

Good idea! :)
It's worth a try and it's a quick test. I'll do it and post the result here.

>2.Pass the values to the HTML template to see if it's working better.

You mean like this https://github.com/shreyanshk/flask-firehose#pushing-resources?

Thanks for the review Karl!

Reply:
>  It's worth a try and it's a quick test. I'll do it and post the result here.

It didn't work. :(

Reply:
so just the `Link` without the code in the template didn't work?
What about option 2?

Reply:
No. :/

I didn't understand what you meant on option 2. You meant I should do something like this `<link rel="stylesheet" href="{{ push('/static/css/main.css', as='style', rel='preload') }}">`? Adding the `preload` tag on the HTML?

Reply:
It's fine!
Already tried that. Same result as adding on the backend. I even tried both at the same time :p
No positive results.